HeterogeneousCore/AlpakaCore/python/functions.py

```diff
@@ -0,0 +1,23 @@
+def makeSerialClone(module, **kwargs):
+    type = module._TypedParameterizable__type
+    if type.endswith('@alpaka'):
+        # alpaka module with automatic backend selection
+        base = type.removesuffix('@alpaka')
+    elif type.startswith('alpaka_serial_sync::'):
+        # alpaka module with explicit serial_sync backend
+        base = type.removeprefix('alpaka_serial_sync::')
+    elif type.startswith('alpaka_cuda_async::'):
+        # alpaka module with explicit cuda_async backend
+        base = type.removeprefix('alpaka_cuda_async::')
+    elif type.startswith('alpaka_rocm_async::'):
+        # alpaka module with explicit rocm_async backend
+        base = type.removeprefix('alpaka_rocm_async::')
```

Comment:
The if-else chain could be replaced with e.g. a regex to remove the `alpaka_[^:]*::` part.